store/LocalStore.js

```diff
@@ -2,14 +2,23 @@
 
 const lru = require('tiny-lru')
 
-function LocalStore (timeWindow, cache) {
-  this.lru = lru(cache || 5000)
-  setInterval(this.lru.clear.bind(this.lru), timeWindow).unref()
+function LocalStore (opts) {
+  this.lru = lru(opts)
+  this.timers = {}
 }
 
-LocalStore.prototype.incr = function (ip, cb) {
-  var current = this.lru.get(ip) || 0
-  this.lru.set(ip, ++current)
+LocalStore.prototype.incr = function (prefix, key, timeWindow, cb) {
+  let keyName = `${prefix}-${key}`
+  let current = this.lru.get(keyName) || 0
+  this.lru.set(keyName, ++current)
+
+  if (!this.timers[keyName]) {
+    this.timers[keyName] = setTimeout(() => {
+      this.lru.delete(keyName)
+      this.timers[keyName] = null
+    }, timeWindow)
```

Comment:
this would need an unref().

The previous approach was way more performant. This is going to create a significant higher amount of timers.. and those are expensive. Could this be refactored to avoid this and keep a global timer? I think keeping a plugin-wide timewindow make sense, otherwise things would become a bit more weird to debug.

I’m also ok on a per-route timewindow.

Reply:
Hello @mcollina ,

Ok .  I will change that.

Reply:
Hello @mcollina ,

actually I'm not sure to understand,  would you mind refactor this part. I added the `unref`

Reply:
@mcollina 
> This is very good work!
> 
> I would also like to retain the previous behavior. By default, the rate limit is applied to all routes. It can be disabled, and then it’s per-route. We should also support keeping it enabled for all routes, but off or with different settings for one.

done. I also updated the doc with what i've done today. 

please validate.